pkg/validator/hcl.go

```diff
@@ -0,0 +1,42 @@
+package validator
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/hclparse"
+)
+
+// HclValidator is used to validate a byte slice that is intended to represent a
+// HashiCorp Configuration Language (HCL) file.
+type HclValidator struct{}
+
+// Validate checks if the provided byte slice represents a valid .hcl file.
+//
+// The hcl parser uses FIFO to determine which error to display to the user. For
+// more information, see the documentation at:
+//
+// https://pkg.go.dev/github.com/hashicorp/hcl/v2#Diagnostics.Error
+//
+// If the hcl.Diagnostics slice contains more than one error, the wrapped
+// error returned by this function will include them as "and {count} other
+// diagnostic(s)" in the error message.
+//
+// If the parsing error does not produce an hcl.Diagnostics slice, a generic
+// error will be returned, wrapping the input file.
+func (hclv HclValidator) Validate(b []byte) (bool, error) {
+	_, diags := hclparse.NewParser().ParseHCL(b, "")
+	if diags == nil {
+		return true, nil
+	}
+
+	if len(diags) == 0 {
+		return false, fmt.Errorf("error validating .hcl file: %w", diags)
```

Comment:
We need a test that covers this case as well as the case mentioned in the comment above where hcl.Diagnostics slice contains more than one error.  I only see the following in the test output.

```
    × /test2/test/fixtures/subdir2/bad.hcl
        error: error at line 1 column 1: :1,1-2: Invalid argument name; Argument names must not be quoted.
```

Reply:
I think it's actually best to remove this block, it should be the same as [`diags == nil`](https://go.dev/play/p/6DPkvnPiT1R) .

Reply:
That's fine to remove but we still need a test that validates your notes about the diags.